test/datetime.test.js

```diff
@@ -0,0 +1,93 @@
+// Copyright IBM Corp. 2012,2016. All Rights Reserved.
+// Node module: loopback-connector-mysql
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var DateString = require('../node_modules/loopback-datasource-juggler/lib/date-string');
+var fmt = require('util').format;
+var should = require('./init.js');
+
+var db, Order, Person;
+describe('MySQL datetime handling', function() {
+  var personDefinition = {
+    name: String,
+    gender: String,
+    married: Boolean,
+    dob: DateString,
```

Comment:
I think after `DateString` get registered to a model, user don't have to specify this type by requiring the file in datasource directly, you can define it by `doc: {type: 'datestring'}` then model builder knows its type right?